components/board/genericlinux/config.go

```diff
@@ -40,3 +41,40 @@ func (conf *Config) Validate(path string) ([]string, error) {
 	}
 	return nil, nil
 }
+
+// UnderlyingConfig is a struct containing absolutely everything a genericlinux board might need
+// configured. It is a union of the configs for the customlinux boards and the genericlinux boards
+// with static pin definitions, because those components all use the same underlying code but have
+// different config types (e.g., only genericlinux has named I2C and SPI buses, while only
+// customlinux can change its pin definitions during reconfiguration). The UnderlyingConfig struct
+// is a unification of the two of them. Whenever we go through reconfiguration, we convert the
+// provided config into this type, and then reconfigure based on this.
+type UnderlyingConfig struct {
```

Comment:
Not sure about this name. Arguably this should be just `Config` and the other struct should be renamed `ConstPinDefsConfig` or something. That would also make the diff ~15 lines shorter in this PR, because so many of the functions that look at configs wouldn't need to change the name of their argument type.

Reply:
yeah, I like that name much more!